pkg/rpc/connection_class.go

```diff
@@ -61,36 +59,18 @@ func (c ConnectionClass) String() string {
 // SafeValue implements the redact.SafeValue interface.
 func (ConnectionClass) SafeValue() {}
 
-var systemClassKeyPrefixes = []roachpb.RKey{
-	roachpb.RKey(keys.Meta1Prefix),
-	roachpb.RKey(keys.NodeLivenessPrefix),
-}
-
-// isSystemKey returns true if the given key belongs to a range eligible for
-// SystemClass connection.
-//
-// Generally, not all system ranges are eligible. For example, the timeseries
-// ranges are not, because they can be busy and disrupt other system traffic. We
-// try to make SystemClass responsive by keeping it small.
-func isSystemKey(key roachpb.RKey) bool {
-	// An empty RKey addresses range 1 and warrants SystemClass.
-	if len(key) == 0 {
-		return true
-	}
-	for _, prefix := range systemClassKeyPrefixes {
-		if bytes.HasPrefix(key, prefix) {
-			return true
-		}
-	}
-	return false
-}
+// All the ranges that include keys between /Min and/ System/tsd should
+// use the system class. To avoid relying on the NoSplit list we just check if
```

Comment:
Its important because the previous logic relied on checking that the range start key starts with a particular prefix, which breaks if the range splits.

Reply:
Anyway I clean up the comment and removed the reference to NoSplit since its not particularly important here.

Reply:
Does it break? If a range with that's entirely under a particular prefix splits, both LHS and RHS will have the same prefix.

I think we are rather relying on the fact that ranges with these particular prefixes don't merge with each other (and so they retain their prefix at at times).

Reply:
I don't know if this conversation is necessarily pertinent to this PR landing as is, but I think splits still matter here. 

The default layout is:

`demo@127.0.0.1:26257/movr> select range_id, start_pretty, end_pretty, replicas, lease_holder, range_size from crdb_internal.ranges where range_id < 10 order by range_id ;
  range_id |      start_pretty       |       end_pretty        |  replicas   | lease_holder | range_size
-----------+-------------------------+-------------------------+-------------+--------------+-------------
         1 | /Min                    | /System/NodeLiveness    | {1,2,3,4,6} |            4 |      76937
         2 | /System/NodeLiveness    | /System/NodeLivenessMax | {1,2,3,4,6} |            4 |      17448
         3 | /System/NodeLivenessMax | /System/tsd             | {2,3,4,5,6} |            4 |     765852
         4 | /System/tsd             | /System/"tse"           | {1,2,5}     |            5 |   13271363
         5 | /System/"tse"           | /Table/0                | {1,2,3,4,5} |            1 |          0
         6 | /Table/0                | /Table/3                | {1,2,3,4,5} |            4 |          0
         7 | /Table/3                | /Table/4                | {2,3,4,5,6} |            6 |      96139
         8 | /Table/4                | /Table/5                | {1,2,3,4,6} |            1 |        493
         9 | /Table/5                | /Table/6                | {2,3,4,5,6} |            6 |        507
(9 rows)`

I got this from running cockroach demo. Any split in range 2 will be captured by the prefix logic, since it will always have the prefix of `/System/NodeLiveness' but that does not hold true for range 1 and 3, since we cant get guess what the prefix is or maintain it.  So in practice both answers are correct here if you have merges the prefix logic breaks, if you have splits it also breaks.

Reply:
I think the key point here is that with the new proposed logic we are not longer dependent on the split or merge logic, since we are just checking if that start key of the range is within a particular span. The logic does not consider if the system space has 1000 ranges or 1. The only weird scenario that is not considered here is that if for example the time series range merges with the range below (i.e. 4 merges with 5) then you would get `/System/NodeLivenessMax - /System/"tse"`, then we are dependent on merges as you describe and the properties of splits/merges become important. Thus if someone changed the static split list this would break. Its probably worth leaving a comment to that effect here.